### PR TITLE
Fix scanFile signature warnings

### PR DESCRIPTION
**Pull request by PVince81**
@icewind1991 @schiesbn @DeepDiver1975 @nickvergessen @LukasReschke 

Fixes annoying warning about signature mismatch of `scanFile()`

### CODE REVIEW

**Comment by nickvergessen** [comment]
:+1:

**Comment by icewind1991** [comment]
:+1: makes sense

File: apps/files_sharing/lib/external/scanner.php [hunk 1]
```diff
@@ -33,7 +33,7 @@ public function scan($path, $recursive = self::SCAN_RECURSIVE, $reuse = -1) {
 	 * @param int $reuseExisting
 	 * @return array an array of metadata of the scanned file
 	 */
-	public function scanFile($file, $reuseExisting = 0) {
+	public function scanFile($file, $reuseExisting = 0, $parentId = -1, $cacheData = null) {
```
Comment (LukasReschke) [inline apps/files_sharing/lib/external/scanner.php]: Also add to PHPDoc? :see_no_evil:

**Comment by karlitschek** [comment]
:+1:

File: apps/files_sharing/lib/external/scanner.php (hunk 1 above)
Comment (PVince81) [inline apps/files_sharing/lib/external/scanner.php]: They're not even used.

Comment (LukasReschke) [inline apps/files_sharing/lib/external/scanner.php]: And...? - EVERY IDE will complain now as will Scrutinizer. PHPDoc mandates that every entry needs get added. Good job :-1:

Comment (LukasReschke) [inline apps/files_sharing/lib/external/scanner.php]: Optional parameters will get hinted by IDEs by the usage of defaults in the function signature.

Comment (LukasReschke) [inline apps/files_sharing/lib/external/scanner.php]: Optional parameters will get hinted by IDEs by the usage of defaults in the function signature.

Comment (LukasReschke) [inline apps/files_sharing/lib/external/scanner.php]: Will create follow-up PR....

Comment (LukasReschke) [inline apps/files_sharing/lib/external/scanner.php]: https://github.com/owncloud/core/pull/14944

**Pull request merged**